src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,86 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import React from 'react';
+
+interface Props {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberPag = Math.ceil(total / perPage);
+
+  function countPagination() {
```

Comment:
It's more efficient to use useMemo for the 'countPagination' function to avoid unnecessary re-rendering.